lib/vcs.ml

```diff
@@ -400,6 +400,14 @@ let tag_exists ~dry_run r tag =
   | (`Git, _, _) as r -> git_tag_exists r ~dry_run tag
   | `Hg, _, _ -> failwith "TODO"
 
+let tag_points_to_commit r tag commit_ish =
```

Comment:
I'm not sure this function is required. What would you think about extracting a function:
```ocaml
(** [tag_points_to t tag] returns the commit hash [tag] points to or [None] if it doesn't exist *)
tag_points_to : dry_run: bool -> t -> string -> string option
```
and implement `tag_exists` on top of it. We can then clean the tag  command code a bit so that it starts by getting the `commit_ish` actual commit. Note that this we should fail if it doesn't exist anyway so might as well do it early. It would then get the commit for the tag and behave accordingly.

Reply:
Yes, that structure you're suggesting is far nicer! I'm just not sure if I understand what you have in mind for `tag_points_to` and `tag_exists`. Could you have another look (once I push the changes)?